### PR TITLE
Model refactor

### PR DESCRIPTION
**Pull request by martincarrera**
#3

### CODE REVIEW

File: src/helpers/asset-helper.js [hunk 1]
```diff
@@ -6,17 +6,13 @@ module.exports = {
     var content = slackmessage.split(' ');
 
     var link = content.shift();
-    var categories = [];
-    content.forEach(function(item, index) {
-      if(item.charAt(0) == '@') {
-        categories.push(item);
-      }
+    var categories = content.filter(function (x) {
```
Comment (ndelvalle) [inline src/helpers/asset-helper.js]: We should remove duplicates

File: src/models/asset-model.js [hunk 2]
```diff
@@ -5,25 +5,41 @@ var assetHelper = require('../helpers/asset-helper');
...
-   }
-   return newSchemaModel.saveAsync().then(function(doc) {
-     return {
-       'response_type': 'in_channel',
-       text: 'The link ' + doc.link + ' was added successfully by ' + doc.createdBy.user.name
-     };
-   });
- }
+  createFromSlack(input) {
+    var self = this;
```
Comment (ndelvalle) [inline src/models/asset-model.js]: Are we using self variable?

File: src/models/category-model.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+var CategoryModel = class extends Model {
+  constructor(SchemaModel) {
+    super(SchemaModel);
+  }
+
+  findOrCreate(category){
+    var self = this;
+    console.log('into findOrCreate with category ' + category);
+    return this.findOne({category: category})
```
Comment (ndelvalle) [inline src/models/category-model.js]: Use self to keep consistency

**Comment by gillchristian** [comment]
@ndelvalle add your changes so I can marge it :wink:

**Comment by ndelvalle** [comment]
@gillchristian done!

**Comment by gillchristian** [comment]
LGTM!

**Pull request merged**